src/main/java/com/code_intelligence/jazzer/mutation/mutator/collection/MapMutatorFactory.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright 2023 Code Intelligence GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.code_intelligence.jazzer.mutation.mutator.collection;
+
+import static com.code_intelligence.jazzer.mutation.support.TypeSupport.parameterTypesIfParameterized;
+import static java.lang.Math.min;
+import static java.util.stream.Collectors.toMap;
+
+import com.code_intelligence.jazzer.mutation.api.*;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.lang.reflect.AnnotatedType;
+import java.util.*;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+final class MapMutatorFactory extends MutatorFactory {
+  @Override
+  public Optional<SerializingMutator<?>> tryCreate(AnnotatedType type, MutatorFactory factory) {
+    // noinspection OptionalIsPresent
+    return parameterTypesIfParameterized(type, Map.class)
+        .map(parameterTypes
+            -> parameterTypes.stream()
+                   .map(factory::tryCreate)
+                   .flatMap(o -> o.isPresent() ? Stream.of(o.get()) : Stream.empty())
+                   .collect(Collectors.toList()))
+        .filter(elementMutators -> elementMutators.size() == 2)
+        .map(elementMutators
+            -> new MapMutator<>(elementMutators.get(0), elementMutators.get(1),
+                MapMutator.DEFAULT_MIN_SIZE, MapMutator.DEFAULT_MAX_SIZE));
+  }
+
+  private static final class MapMutator<K, V> extends SerializingInPlaceMutator<Map<K, V>> {
+    private static final int DEFAULT_MIN_SIZE = 0;
+    private static final int DEFAULT_MAX_SIZE = 1000;
```

Comment:
Can users increase this size without building their own Jazzer, if for some reason they need `> DEFAULT_MAX_SIZE` keys in their map?

Reply:
Good point. I added support for the `@WithSize` annotation to set min/max freely.

Reply:
An observation: By itself, the fuzzer cannot generate maps with more keys than `maxSize`.
However, at the moment `maxSize` is not checked when reading a map with more keys than `maxSize`.
That means that should our corpus contain inputs with larger maps from previous runs, even if we set the `maxSize` to some value, those inputs will currently be loaded by the mutator framework.
Depending on what the user wants, this might get confusing.

In libFuzzer, inputs larger than `-max_len` are usually ignored.

Maybe this is something we can address in the future.

Reply:
Good catch! We should address this now as changing the `read`/`write` methods is a breaking change. We could just use the min/max with the configured min/max sizes there. I think that e.g. the integer mutator is already doing that.

Reply:
In that case `ListMutatorFactory` should be updated as well.

Is the strategy then to read all elements, but to keep only the `maxSize`; and fill up to 'minSize` if there are not enough elements? Or should such inputs be discarded altogether? In both cases only the `read` method should be changed.

The issue with the always read strategy is that reproducing a finding found with specific size settings is now tied to the input as well as to the configured sizes.
If the user changes the size configuration, a finding might stop being a finding until they change the sizes back.

During fuzzing, libFuzzer ignores inputs greater than `max_len` and won't generate inputs greater than that; but when executing the fuzzer with specific inputs, libFuzzer ignores the `max_len` setting and uses all available data.

I think, we should do the same---inputs not compliant with configured sizes should be ignored during fuzzing, but loaded during the reproducing phase. What do you think?